### DISCUSSION TITLE
Adaptive Rho by the Solution

### DISCUSSION THREAD

**Discussion by RoyiAvital** [osqp/osqp#551] [General]
Hello,

In the paper (Great paper! I love that you share low level details so no need to even look at the code) you say that the adaptive update of `rho` is very important.
You also define the perfect case for its values according to the active set of constraint.

I wonder, why don't you update the matrix according to the actual active sets?
After few iterations you start to get some idea so you can do better than what you defined at the beginning.

**Comment by gbanjac** [comment]
Updating `rho` is computationally expensive as it involves refactoring the KKT matrix. We thus don't want to update `rho` often.

**Comment by RoyiAvital** [comment]
I understand. I am saying that when you do, why not build it with some info about the active constraint?
It could also be like another "Polish". Like after there is a convergence, update the matrix with the same trick as in the polish step and do 10% / 5% more iterations.
